core/src/main/java/org/testcontainers/containers/SocatContainer.java

```diff
@@ -15,8 +16,16 @@
 
     private final Map<Integer, String> targets = new HashMap<>();
 
+    /**
+     * @deprecated use {@link SocatContainer(DockerImageName)} instead
+     */
+    @Deprecated
```

Comment:
Just thinking... what if we keep the default constructors in "utility" classes like this one, where the users do not / should not care about the version used? it is a bit contradictory to the idea of this whole change, but I _think_ a few exceptions can be made

Reply:
Yeah, I was thinking the same. If these had not been essentially in our public API it would have been a no-brainer.

But I'm fine with doing it. I'll move to deprecate these classes away from view (e.g. if we can eventually get them to be package-private then this becomes moot).

Reply:
well, I would not hide/deprecate them, because they are helpful for building complex 3rd party containers

Reply:
Reverted